core/src/test/java/org/elasticsearch/search/indicesboost/SimpleIndicesBoostSearchIT.java

```diff
@@ -77,31 +75,154 @@ public void testIndicesBoost() throws Exception {
+
+        client().index(indexRequest("multi_test1").type("type1").id("1")
+            .source(jsonBuilder().startObject().field("test", "value check").endObject())).actionGet();
+        client().index(indexRequest("multi_test2").type("type1").id("1")
+            .source(jsonBuilder().startObject().field("test", "value beck").endObject())).actionGet();
+        refresh();
+
+        float aliasBoost = 1.1f;
+        float wildcardBoost = 0.9f;
+
+        logger.info("Query with multi_test1 boosted");
+        SearchResponse response = client().search(searchRequest()
+            .searchType(SearchType.QUERY_THEN_FETCH)
+            .source(searchSource()
+                .explain(true)
+                .indexBoost("alias1", aliasBoost)
+                .indexBoost("*_test1", wildcardBoost)
+                .query(termQuery("test", "value")))
+        ).actionGet();
+
+        assertThat(response.getHits().totalHits(), equalTo(2L));
+        logger.info("Hit[0] {} Explanation {}", response.getHits().getAt(0).index(), response.getHits().getAt(0).explanation());
+        logger.info("Hit[1] {} Explanation {}", response.getHits().getAt(1).index(), response.getHits().getAt(1).explanation());
+        assertThat(response.getHits().getAt(0).index(), equalTo("multi_test1"));
+        assertThat(response.getHits().getAt(1).index(), equalTo("multi_test2"));
+
+        logger.info("Query with multi_test1 de-boosted");
+        response = client().search(searchRequest()
+            .searchType(SearchType.QUERY_THEN_FETCH)
+            .source(searchSource()
+                .explain(true)
+                .indexBoost("*_test1", wildcardBoost)
+                .indexBoost("alias1", aliasBoost)
+                .query(termQuery("test", "value")))
+        ).actionGet();
+
+        assertThat(response.getHits().totalHits(), equalTo(2L));
+        logger.info("Hit[0] {} Explanation {}", response.getHits().getAt(0).index(), response.getHits().getAt(0).explanation());
+        logger.info("Hit[1] {} Explanation {}", response.getHits().getAt(1).index(), response.getHits().getAt(1).explanation());
+        assertThat(response.getHits().getAt(0).index(), equalTo("multi_test2"));
+        assertThat(response.getHits().getAt(1).index(), equalTo("multi_test1"));
+    }
+
+    public void testNonExistentIndexAliasBoost() throws Exception {
+        createIndex("test1");
+        ensureGreen();
+
+        client().index(indexRequest("test1").type("type1").id("1")
+            .source(jsonBuilder().startObject().field("test", "value check").endObject())).actionGet();
+        refresh();
+
+        float boost = 1.1f;
+
+        logger.info("Query with non existent index/alias boost");
+        SearchResponse response = client().search(searchRequest()
+            .searchType(SearchType.QUERY_THEN_FETCH)
+            .source(searchSource()
+                .explain(true)
+                .indexBoost("non_existent", boost)
```

Comment:
I see why you used lenient indices options, so that nothing breaks if non existing indices are specified. But maybe that should be configurable like I commented above.